src/Content/Version.php

```diff
@@ -283,6 +293,18 @@ protected function prepareFieldsAfterRead(array $fields, Language $language): ar
 		return $this->convertFieldNamesToLowerCase($fields);
 	}
 
+	/**
+	 * Make sure that the Content object receives the right set of fields
+	 * filtering fields used for lower logic (e.g. lock)
+	 */
+	protected function prepareFieldsForContent(
+		array $fields,
+		Language $language
+	): array {
+		unset($fields['lock'], $fields['slug']);
```

Comment:
Should we make this dependent on the model and filter out sort and template here for files instead? Slug also only needs to be filtered out for pages.

Reply:
Yes, changing it.

Only that `sort` currently is really treated as content field in the way how we use it. So removing that breaks a few things and would likely be a breaking change for users, which we could do for v5. What do you think? (We would need to create a dedicated `File::sort(): Field` method to keep the breaking change small).

Reply:
Ok, let's take this to another PR and make sure we get it right there. I think we should really start treating such values as system values and not mix them up with fields. The compatibility method sounds like a good plan, but I would probably immediately deprecate it.

Reply:
Added a "nice to have" todo in the Notion Changes todo list.